pkg/id/id.go

```diff
@@ -94,19 +94,21 @@ func NewAllocator(params *AllocatorParams) Allocator {
 }
 
 // Alloc returns a new id.
-func (alloc *allocatorImpl) Alloc() (uint64, error) {
+func (alloc *allocatorImpl) Alloc(count uint32) (uint64, uint32, error) {
 	alloc.mu.Lock()
 	defer alloc.mu.Unlock()
 
-	if alloc.base == alloc.end {
-		if err := alloc.rebaseLocked(true); err != nil {
-			return 0, err
+	for range count {
+		if alloc.base == alloc.end {
+			if err := alloc.rebaseLocked(true); err != nil {
```

Comment:
It may generate many etcd requests.

Reply:
Only when `alloc.base == alloc.end`, it will rebase.

Reply:
What about just calculating if the rebase is necessary rather than passively waiting for the loop to do it?